src/org/thoughtcrime/securesms/WebxdcActivity.java

```diff
@@ -332,6 +339,28 @@ public boolean sendStatusUpdate(String payload, String descr) {
       return true;
     }
 
+    @JavascriptInterface
+    public boolean sendMessage(String payload) {
+      Log.i(TAG, "sendMessage");
+      try {
+        JSONObject jsonObject = new JSONObject(payload);
+        byte[] data = Base64.decode(jsonObject.getString("__blobBase64"), Base64.NO_WRAP | Base64.NO_PADDING);
+        String[] fileName = jsonObject.getString("fileName").split("\\.");
```

Comment:
this looks dangerous, maybe use https://docs.oracle.com/javase/6/docs/api/java/lang/String.html#split%28java.lang.String,%20int%29 to split at most in two strings, and probably also check the case where the filename has no dot "." (no extension) at all